### PR TITLE
WIP/MNT: Update SWE to use packet_file_to_datasets

### PR DESCRIPTION
**Pull request by greglucas**
# Change Summary

## Overview

This is switching SWE over to packet_file_to_datasets rather than iterating through all of the packets individually.

### CODE REVIEW

File: imap_processing/swe/l1a/swe_science.py [hunk 1]
```diff
@@ -202,7 +187,7 @@ def swe_science(decom_data: list, data_version: str) -> xr.Dataset:
     # Add APID to global attrs for following processing steps
     l1a_global_attrs = cdf_attrs.get_global_attributes("imap_swe_l1a_sci")
     # Formatting to string to be complaint with ISTP
-    l1a_global_attrs["packet_apid"] = f"{decom_data[0].header['PKT_APID'].raw_value}"
+    # l1a_global_attrs["packet_apid"] = SWEAPID.SWE_SCIENCE.value
```
Comment (tech3371) [inline imap_processing/swe/l1a/swe_science.py]: can you uncomment this?

File: imap_processing/swe/l1a/swe_science.py [hunk 2]
```diff
@@ -215,16 +200,23 @@ def swe_science(decom_data: list, data_version: str) -> xr.Dataset:
...
+            "type",
+            "sec_hdr_flg",
+            "pkt_apid",
+            "seq_flgs",
+            "src_seq_ctr",
+            "pkt_len",
+        ]
+    )
+    for var_name, arr in l0_dataset.variables.items():
+        arr.attrs = cdf_attrs.get_variable_attributes(var_name)
```
Comment (tech3371) [inline imap_processing/swe/l1a/swe_science.py:219]: looks like we got the CDF attrs set properly for remaining variables. Did you want me to look into CDF attrs for another place/case after this PR?

File: imap_processing/swe/packet_definitions/swe_packet_definition.xml
Comment (tech3371) [inline imap_processing/swe/packet_definitions/swe_packet_definition.xml:1]: Thank you for doing this! This was on my todo list.

File: imap_processing/tests/swe/test_swe_l1a_science.py [hunk 3]
```diff
@@ -29,22 +29,20 @@ def test_swe_raw_science_data(decom_test_data):
...
-            continue
-        if key == "SCIENCE_DATA":
-            continue
-        # check if the data is the same
-        assert value.raw_value == validation_data[key]
+    # compare raw values of the packets
+    shared_keys = set([x.lower() for x in validation_data.keys()]).intersection(
+        first_data.keys()
+    )
+    # TODO: Why are all the fields not the same between the two
```
Comment (tech3371) [inline imap_processing/tests/swe/test_swe_l1a_science.py:39]: Can you clarify this?

File: imap_processing/tests/swe/test_swe_l1a_science.py [hunk 4]
```diff
@@ -68,24 +66,20 @@ def test_swe_derived_science_data(decom_test_data):
...
 
 
 def test_data_order(decom_test_data):
     # test that the data is in right order
-    assert decom_test_data[0].data["QUARTER_CYCLE"].derived_value == "FIRST"
-    assert decom_test_data[1].data["QUARTER_CYCLE"].derived_value == "SECOND"
-    assert decom_test_data[2].data["QUARTER_CYCLE"].derived_value == "THIRD"
-    assert decom_test_data[3].data["QUARTER_CYCLE"].derived_value == "FORTH"
+    np.testing.assert_array_equal(
+        decom_test_data.isel(epoch=slice(0, 4))["quarter_cycle"], [0, 1, 2, 3]
```
Comment (tech3371) [inline imap_processing/tests/swe/test_swe_l1a_science.py:75]: nice!

**Review by tech3371 — COMMENTED** [review]
Looks great! Thank you for doing this work!

**Review by tech3371 — APPROVED** [review]

File: imap_processing/tests/swe/test_swe_l1a_science.py (hunk 3 above)
Comment (greglucas) [reply imap_processing/tests/swe/test_swe_l1a_science.py:39]: I'm curious why all the validation data isn't the same as the dataset we are producing. I would think that our dataset should contain all the values that are in the validation data, but there is a difference between the validation data keys, and the created dataset keys.

These are the variables that are in one or the other, but not both.
```
{'cem1', 'phverno', 'cem3', 'phapid', 'phgroupf', 'cem2', 'phtype', 'phseqcnt', 'timestamp', 'cem5', 'cem4', 'cem6', 'cem7', 'phdlen', 'phshf'}
```

Comment (tech3371) [reply imap_processing/tests/swe/test_swe_l1a_science.py:39]: I see. Those are CCSDS header but they called it different than our standard names in XTCE. They called it like this.
```
PHVERNO, PHTYPE, PHSHF, PHAPID, PHGROUPF, PHSEQCNT, PHDLEN
```

and remaining are decommed data of `SCIENCE_DATA` -
```
CEM1, CEM2, CEM3, CEM4, CEM5, CEM6, CEM7
```

Comment (greglucas) [reply imap_processing/tests/swe/test_swe_l1a_science.py:39]: OK, would you like me to remove the comment or update it to something else?

Comment (tech3371) [reply imap_processing/tests/swe/test_swe_l1a_science.py:39]: If you can add comment to validate those CEM# data when we get better data, that would be great.

Comment (greglucas) [reply imap_processing/tests/swe/test_swe_l1a_science.py:39]: :+1: Sounds good. I just updated the comment.

**Pull request merged**